apps/system/js/places.js

```diff
@@ -323,6 +387,17 @@
       var lastTopSite = this.topSites[numTopSites - 1];
       if (numTopSites < this.MAX_TOP_SITES ||
         place.frecency > lastTopSite.frecency) {
+        var newTopSites = [];
+        this.topSites.forEach(e => {
+          if (!e) {
+            return;
+          }
+
+          if (e.url !== place.url) {
+            newTopSites.push(e);
+          }
+        });
+        this.topSites = newTopSites;
```

Comment:
If we can remove the `if (!e) {` I would prefer that, its defensive programming and we shouldnt ever get to a situation where there is a null value, but otherwise looks good, I do think a comment in here just to explain what it is doing would make things a little clearer.